packages/nvidia_nat_a2a/src/nat/plugins/a2a/client/client_base.py

```diff
@@ -0,0 +1,328 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import logging
+from collections.abc import AsyncGenerator
+from datetime import timedelta
+from uuid import uuid4
+
+import httpx
+
+from a2a.client import A2ACardResolver
+from a2a.client import Client
+from a2a.client import ClientConfig
+from a2a.client import ClientEvent
+from a2a.client import ClientFactory
+from a2a.types import AgentCard
+from a2a.types import Message
+from a2a.types import Part
+from a2a.types import Role
+from a2a.types import Task
+from a2a.types import TextPart
+
+logger = logging.getLogger(__name__)
+
+
+class A2ABaseClient:
+    """
+    Minimal A2A client for connecting to an A2A agent.
+
+    Args:
+        base_url: The base URL of the A2A agent
+        task_timeout: Timeout for task operations (default: 300 seconds)
+    """
+
+    def __init__(
+            self,
+            base_url: str,
+            agent_card_path: str = "/.well-known/agent-card.json",
```

Comment:
Should this be `"./well-known/agent-card.json"`?

Reply:
The dir is ".well-known" :) 
```
$ curl $A2A_SERVER_URL/.well-known/agent-card.json | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1205  100  1205    0     0   3373      0 --:--:-- --:--:-- --:--:--  3403
{
  "capabilities": {
    "pushNotifications": false,
    "streaming": true
  },
```